alonzo/formal-spec/txinfo.tex

```diff
@@ -3,5 +3,232 @@ \section{TxInfo Construction}
+    & \mathsf{Cert} & \type{P.Certifying} \\
+    & \mathsf{Rwrd} & \type{P.Rewarding} \\
+    & \mathsf{Mint} & \type{P.Minting} \\
+    & \mathsf{Spend} & \type{P.Spending} \\
+    ~\\
+    & \emph{Execution Budget Types} \\
+    & \Integer & \type{P.ExCPU} \\
+    & \Integer  & \type{P.ExMemory}\\
+    & \ExUnits & \type{P.ExBudget}  \\
+    ~\\
+    & \emph{TxInfo} \\
+    & \TxInfo = \\
+      & ~~~~ \seqof{\type{P.TxInInfo^?}} \\ % & \text{inputs} \\ %\type{P.txInfoInputs} &
+      & ~~~~ \text{$\uparrow$ Realized inputs (built by $\fun{txInfoIn}$)} \\
+      & ~~~~ \times \seqof{\type{P.TxOut^?}}  \\
+      & ~~~~ \text{$\uparrow$ outputs} \\ %\type{P.txInfoOutputs}
+      & ~~~~ \times \type{P.Value} \\
+      & ~~~~ \text{$\uparrow$ fee, built by applying $\fun{transValue}\circ\fun{inject}$ to the $\Coin$ fee} \\ %= transValue (inject @(Mary.Value (Crypto era)) fee), \type{P.txInfoFee}
+      & ~~~~ \times \type{P.Value}  \\
+      & ~~~~ \text{$\uparrow$ mint field, built from the mint $\Value$ by $\fun{transValue}$} \\ % = transValue forge, \type{P.txInfoMint}
+      & ~~~~ \times \seqof{\type{P.DCert}}  \\
+      & ~~~~ \text{$\uparrow$ list of certificates} \\ %= foldr (\c ans \to transDCert c : ans) [] (certs' tbody), \type{P.txInfoDCert}
+      & ~~~~ \times (\type{P.StakingCredential} \times \Integer) \\
+      & ~~~~ \text{$\uparrow$ the $\Wdrl$s, reward withdrawal pairs of credential and reward amount} \\ % = Map.toList (transWdrl (wdrls' tbody)), \type{P.txInfoWdrl}
+      & ~~~~ \times \type{P.POSIXTimeRange} \\
+      & ~~~~ \text{$\uparrow$ transaction validity POSIX time range (built by $\fun{transVITime}$)} \\ % = timeRange, \type{P.txInfoValidRange}
+      & ~~~~ \times \seqof{\type{P.PubKeyHash}}  \\
+      & ~~~~\text{$\uparrow$ hashes of keys whose signature are required} \\ % = map transKeyHash (Set.toList (reqSignerHashes' tbody)), \type{P.txInfoSignatories}
+      & ~~~~ \times \seqof{(\DataHash \times \Data)}  \\
+      & ~~~~ \text{$\uparrow$ list of pairs of datum hashes and their datum preimages} \\ % = map transDataPair datpairs, \type{P.txInfoData}
+      & ~~~~ \times \type{P.TxId} \\ % =  (transSafeHash (hashAnnotated @(Crypto era) tbody)) \type{P.txInfoId}
+      & ~~~~ \text{$\uparrow$ list of pairs of datum hashes and their datum preimages}
+  \end{align*}
+  \caption{TxInfo and Constituent Types}
+  \label{fig:txinfo-types-two}
+\end{figure*}
+
+\begin{figure*}[htb]
+  \begin{align*}
+    & \fun{transAddr} \in \Addr  \to  \type{P.Address} \\
+    & \fun{transAddr} ~a = \begin{cases}
+      \Nothing & \text{if}~a~\in~\AddrBS \\
+      (ob,~st) & \text{if}~a~=~(\wcard, ob_P, st_P)
+    \end{cases} \\
+    & \text{Address translation}
+    \nextdef
+    & \fun{transValue} \in \Value \to \type{P.Value} \\
+    & \fun{transValue}~ (c,~ mp) = \{~ pid_P\mapsto (aid_P \mapsto q_P) ~\mid~ pid \mapsto (aid \mapsto q) ~\in ~mp ~\} \\
+    & ~~~~\cup \{~\type{P.adaSymbol}\mapsto (\type{P.adaToken} \mapsto c) ~\} \\
```

Comment:
So Plutus has a pid and asset name for Ada? We should mention that in the MA spec!

Reply:
maybe we should do that in the other (Value) PR?